Source/Observable+QueueSubscribeOn.swift

```diff
@@ -25,51 +25,39 @@ import RxSwift
 
 /// Queue which is used for queueing subscriptions for queueSubscribeOn operator.
 class SerializedSubscriptionQueue {
-    let scheduler: ImmediateSchedulerType
-    let lock = NSLock()
-
     /// First element on queue is curently subscribed and not completed
     /// observable. All others are queued for subscription when the first
     /// one is finished.
     var queue: [DelayedObservableType] = []
 
-    /// Creates a queue in which subscriptions will be executed sequentially after previous ones have finished.
-    /// - parameter scheduler: Scheduler on which subscribption will be scheduled
-    init(scheduler: ImmediateSchedulerType) {
-        self.scheduler = scheduler
-    }
-
-    /// Queue subscription for a queue. If observable is inserted
+    /// Queue subscription. If observable is inserted
     /// into empty queue it's subscribed immediately. Otherwise
     /// it waits for completion from other observables.
     func queueSubscription(observable: DelayedObservableType) {
-        lock.lock(); defer { lock.unlock() }
         let execute = queue.isEmpty
         queue.append(observable)
         if execute {
             // Observable is scheduled immidiately
-            queue.first?.delayedSubscribe(on: scheduler)
+            queue.first?.delayedSubscribe()
         }
     }
 
     func unsubscribe(observable: DelayedObservableType) {
-        lock.lock(); defer { lock.unlock() }
```

Comment:
Are you sure that unsubscribing two scans from different threads will not change the `index` of the `observable`? Unlikely but seems possible

Reply:
Queueing seems to need synchronization too